### DISCUSSION TITLE
Default docker-compose file can't change password without changing the volume's name

### DISCUSSION THREAD

**Discussion by garykuok** [requarks/wiki#4603] [General]
Hello,
I'm new to wiki.js. Recently I've been working on deploying wiki.js with docker-compose. I used the official example from ```https://docs.requarks.io/install/docker```.
After I changed the database password in db and wiki section and start with ```docker-compose up --build```, it'll raise an error saying password authentication error. If I use the exact same file as in example, wiki.js is able to start.
Eventually, I noticed that if I change the volume's name other than ```db_data```, then I can change to whatever password I want and cat start wiki.js without any problem.
I want to ask, what is the root cause of this behavior? Every time when I build the images, I will always do ```docker system prune -a``` to make sure it doesn't use the old data.

I am running wiki.js on a VM with ubuntu:18.04. I only installed Docker and docker-compose on the machine.

Best,
Gary

**Comment by ieaster1** [comment]
This is the expected behavior of the postgres container when it is initialized.  The environment variables are only used when the data volume is empty.

If you'd like to change those variables then you would need to do so manually, clear the data volume, or create a new one.

See: https://hub.docker.com/_/postgres

> Warning: the Docker specific variables will only have an effect if you start the container with a data directory that is empty; any pre-existing database will be left untouched on container startup.

**Reply by garykuok** [reply]
Thank you for your reply. Now I understand this behavior.
